### ISSUE TITLE
Question: About special handling of 'value'-attribute in @Component and @Qualifier

### ISSUE THREAD

**Issue by jbellmann** [sbrannen/spring-composed#19]
You mentioned [this wiki-page](https://github.com/spring-projects/spring-framework/wiki/Spring-Annotation-Programming-Model) and I would be interested in the last point. Is this already documented somewhere?

**Comment by sbrannen** [comment]
The `value` attributes in `@Component` (and related stereotypes) and `@Qualifier` are documented in the corresponding Javadoc for each annotation as well as in the reference manual.
- http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#beans-autowired-annotation-qualifiers
- http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#beans-scanning-name-generator
  - this section unfortunately refers to a `name` value instead of a _name_ `value`.

So that hopefully sheds some light on the story; however, to my knowledge, the special handling of the `value` attributes for such cases is not documented with regard to _composed annotations_ and attribute aliases or overrides. 

In summary, the `value` attributes in `@Component` (and related stereotypes) and `@Qualifier` _cannot_ be influenced by `@AliasFor`. The reason is that the special handling of these `value` attributes was in place years before `@AliasFor` was invented. Consequently, due to backward compatibility issues it is simply not possible to use `@AliasFor` with such `value` attributes.

**Comment by sbrannen** [comment]
FYI: I added this answer to a new FAQ that I just introduced.

https://github.com/spring-projects/spring-framework/wiki/Spring-Annotation-Programming-Model#1-can-aliasfor-be-used-with-the-value-attributes-for-component-and-qualifier

**Comment by jbellmann** [comment]
Vielen Dank.
